k8s/plugin/src/main.rs

```diff
@@ -20,12 +20,22 @@ struct CliArgs {
 
     #[clap(flatten)]
     args: resources::CliArgs,
+
+    /// Use namespace from the current `kubeconfig` context,
+    /// Takes precedence over --namespace or -n flag.
+    #[clap(global = true, long, short = 'c', default_value = "false")]
```

Comment:
We should hide this, since it's not meant to be used for now IIRC

```suggestion
    #[clap(global = true, hide = true, long = "namespace-from-context", default_value = "false")]
```

Reply:
Also should make it conflict with --namespace? Both supplied at the same time doesn't make sense